### ISSUE TITLE
Processing `overrides` rules is not idempotent

### ISSUE THREAD

**Issue by bmish** [ember-template-lint/ember-template-lint#1412]
### TLDR

The processing of `overrides` rules is not idempotent. The `overrides` rules config is corrupted when creating multiple instances of the Linter class with the same `configPath`.

### My config

My .template-lintrc.js looks like this:

```js
module.exports = {
  extends: ['recommended', 'stylistic'],

  rules: {
    'no-bare-strings': true,
    'require-button-type': false,
  },

  overrides: [
    {
      files: ['**/tests/dummy/**/*.hbs'],
      rules: {
        'no-bare-strings': false,
      },
    },
  ],
};
```

### My use case

In my git precommit hook, I am linting each changed file in my monorepo, with a separate instance of the linter class per file (so that each file can use the .template-lintrc.js for its own package).

```js
const TemplateLinter = require('ember-template-lint');

templates.forEach((templateName) => {
  // Find template lint config file for the package this file is in.
  const configPath = findUp.sync('.template-lintrc.js', { cwd: templateName });
  if (!configPath) {
    // Ignore file if we can't find a config file for it.
    return;
  }

  const linter = new TemplateLinter({ configPath });
  const templateContents = fs.readFileSync(templateName, { encoding: 'utf8' });
  const results = linter.verify({ source: templateContents, filePath: templateName });
});
```

As a workaround to get my git commit hook to work, I can ensure that I only create one linter class per `configPath` (by storing a map of `configPath` to linter instance).

### Observed incorrect behavior

When I run the precommit hook on template files in the dummy test app, I get an error that the rule has an invalid config:

```
  -:-  error  The no-bare-strings rule accepts one of the following values.
  * boolean - `true` to enable / `false` to disable,  * array -- an array of strings to whitelist,  * object -- An object with the following keys:,    * `whitelist` -- An array of whitelisted strings,    * `globalAttributes` -- An array of attributes to check on every element,    * `elementAttributes` -- An object whose keys are tag names and value is an array of attributes to check for that tag name
You specified `{"severity":0,"config":false}`  undefined
```

The override rule config gets more and more corrupted each additional time the linter class is created using the same `configPath`:

```
You specified `{"severity":2,"config":{"severity":2,"config":{"severity":0,"config":false}}}`  undefined
```

Part of the problem is that the imported config file is being directly mutated.

### Failing test case

Failing `get-config-test.js` test case that you can add to demonstrate this:

```js
it('processing config overrides is idempotent', function () {
  let config = {
    overrides: [
      {
        files: ['some-path'],
        rules: {
          bar: true,
        },
      },
    ],
  };

  let expectedOverrides = [{ files: ['some-path'], rules: { bar: { config: true, severity: 2 } } }];

  let processedConfig = getProjectConfig({ config });
  expect(processedConfig.overrides).toEqual(expectedOverrides);

  let reprocessedConfig = getProjectConfig({ config });
  expect(reprocessedConfig.overrides).toEqual(expectedOverrides);
});
```

CC: @suchitadoshi1987 @dcyriller 

Related to: https://github.com/ember-template-lint/ember-template-lint/pull/1407
